app/jobs/google/search_progress_job.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Google
+  class SearchProgressJob < ApplicationJob
+    queue_as :default
+
+    def perform(user_id)
+      user = User.find(user_id)
+
+      raise ArgumentError unless user
```

Comment:
I think we don't need it. If the user is not found it will raise `ActiveRecord::RecordNotFound` exception
```suggestion
```